projects/cds/src/profiletag/services/spartacus-events.spec.ts

```diff
@@ -0,0 +1,174 @@
+import { TestBed } from '@angular/core/testing';
+import {
+  Event as NgRouterEvent,
+  NavigationEnd,
+  NavigationStart,
+  Router,
+} from '@angular/router';
+import { Cart, CartService, ConsentService, OrderEntry } from '@spartacus/core';
+import { BehaviorSubject, ReplaySubject } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { CdsConfig } from '../../config';
+import { SpartacusEventTracker } from './spartacus-events';
+
+describe('SpartacusEventTracker', () => {
+  let spartacusEventTracker: SpartacusEventTracker;
+  let getConsentBehavior;
+  let isConsentGivenValue;
+  let routerEventsBehavior;
+  let router;
+  let consentsService;
+  let cartService;
+  let orderEntryBehavior;
+  let cartBehavior;
+  const mockCDSConfig: CdsConfig = {
+    cds: {
+      consentTemplateId: 'PROFILE',
+    },
+  };
+  function setVariables() {
+    getConsentBehavior = new BehaviorSubject<Object>([{}]);
+    isConsentGivenValue = false;
+    routerEventsBehavior = new BehaviorSubject<NgRouterEvent>(
+      new NavigationStart(0, 'test.com', 'popstate')
+    );
+    orderEntryBehavior = new ReplaySubject<OrderEntry[]>();
+    cartBehavior = new ReplaySubject<Cart>();
+    consentsService = {
+      getConsent: () => getConsentBehavior,
+      isConsentGiven: () => isConsentGivenValue,
+    };
+    router = {
+      events: routerEventsBehavior,
+    };
+    cartService = {
+      getEntries: () => orderEntryBehavior,
+      getActive: () => cartBehavior,
+    };
+  }
+  beforeEach(() => {
+    setVariables();
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: Router, useValue: router },
+        {
+          provide: ConsentService,
+          useValue: consentsService,
+        },
+        {
+          provide: CartService,
+          useValue: cartService,
+        },
+        {
+          provide: CdsConfig,
+          useValue: mockCDSConfig,
+        },
+      ],
+    });
+    spartacusEventTracker = TestBed.get(SpartacusEventTracker);
```

Comment:
Should be `spartacusEventTracker = TestBed.get(SpartacusEventTracker as Type<SpartacusEventTracker>);`